### DISCUSSION TITLE
Filter specific genes from the Giotto object

### DISCUSSION THREAD

**Discussion by IgLab-UNAV** [giotto-suite/Giotto#1048] [General]
Sorry for such a basic question, but I've been searching through all the functions and also in this site, and I haven´t found an answer to this. How can I remove features (that from the object once it has been created? I have tried doing it in the @feat_metadata[["cell"]][["rna"]]@metaDT slot, but that doesn´t seem to update @expression$cell$rna$raw@exprMat. I can remove them from the latter, but that will leave the former unaltered. Do I have to change both and that will be enough? Is there a more clean method to do it? Cheers,

**Comment by RubD** [comment]
Not sure what you're trying to do without a reproducible example but [subsetGiotto](https://drieslab.github.io/GiottoClass/reference/subsetGiotto.html) should work for you. You just need to specify which features (e.g. genes) you want to keep.

**Reply by IgLab-UNAV** [reply]
OK, that sounds reasonable, albeit somewhat counterintuite. It works with subsetGiotto(object, feat_ids = keep), where _keep_ is the list of all gene names in the object except for those i want to remove (even if it is only one or two!). Thanks a lot,
